emission/analysis/section_features.py

```diff
@@ -15,18 +15,40 @@
 
 
 # The speed is in m/s
-def calSpeed(trackpoint1, trackpoint2):
+def calSpeed(section):
   from dateutil import parser
-  distanceDelta = calDistance(trackpoint1['track_location']['coordinates'],
-                              trackpoint2['track_location']['coordinates'])
-  timeDelta = parser.parse(trackpoint2['time']) - parser.parse(trackpoint1['time'])
+
+  if ((not section['start_loc']) or (not section['end_loc'])):
```

Comment:
this is an incorrect port. the previous implementation computed the speed between two points along the trip (e.g. two `emission/core/wrapper/location.py` objects or their subclasses. the new implementation computes the speed between the start and end points, which is not equal to the speed along the trajectory if the trip is not in a straight line, and it rarely is.